src/crimson/os/seastore/btree/fixed_kv_btree.h

```diff
@@ -1020,14 +1025,16 @@ class FixedKVBtree {
     assert(is_lba_backref_node(e->get_type()));
     
     auto do_rewrite = [&](auto &fixed_kv_extent) {
-      auto n_fixed_kv_extent = c.cache.template alloc_new_extent<
+      auto extents = c.cache.template alloc_new_extents<
         std::remove_reference_t<decltype(fixed_kv_extent)>
         >(
         c.trans,
         fixed_kv_extent.get_length(),
         fixed_kv_extent.get_user_hint(),
         // get target rewrite generation
         fixed_kv_extent.get_rewrite_generation());
+      ceph_assert(extents.size() == 1);
```

Comment:
Seems only the allocation interface is changed to allow to return 1 or more extents, but all the users still expecting/asserting that the allocation must return only 1 extent ?

Reply:
> Um, Not really. At present, I only allow ObjectDataBlock to be allocated in non-consecutive space.

Ah, OK. Please add this to the commit message and PR desc.

> Others may also be allowed in the future.

It would be case-by-case in the future, I suppose.

How about keeping the original alloc-single path(s), and introduce a dedicated maybe-alloc-multiple path at the according levels to support ObjectDataBlock (probably they can reuse some common parts), so that there are less changes to currently unrelated components -- which seems to be distracting.

Reply:
> It would be case-by-case in the future, I suppose.
> 
> How about keeping the original alloc-single path(s), and introduce a dedicated maybe-alloc-multiple path at the according levels to support ObjectDataBlock (probably they can reuse some common parts), so that there are less changes to currently unrelated components -- which seems to be distracting.

Yes, will do.